pkg/manager/member/tidb_discovery_manager.go

```diff
@@ -130,7 +130,8 @@ func getTidbDiscoveryDeployment(tc *v1alpha1.TidbCluster) (*appsv1.Deployment, e
 				Spec: corev1.PodSpec{
 					ServiceAccountName: meta.Name,
 					Containers: []corev1.Container{{
-						Name: "discovery",
+						Name:      "discovery",
+						Resources: controller.ContainerResource(tc.Spec.Discovery.ResourceRequirements),
```

Comment:
What happen if we don't set `Discovery ResourceRequirements`?

Reply:
By manual test, deployment/basic-discovery for controller TidbCluster/basic was successfully created when I don't set resources requirements.